src/Core/src/Hosting/AppHostBuilder.cs

```diff
@@ -18,29 +18,34 @@ public class AppHostBuilder
 		readonly List<Action<HostBuilderContext, IFontCollection>> _configureFontsActions = new List<Action<HostBuilderContext, IFontCollection>>();
 		readonly List<IConfigureContainerAdapter> _configureContainerActions = new List<IConfigureContainerAdapter>();
 		readonly Func<IServiceCollection> _serviceColectionFactory = new Func<IServiceCollection>(() => new MauiServiceCollection());
-		IServiceFactoryAdapter _serviceProviderFactory = new ServiceFactoryAdapter<IServiceCollection>(new MauiServiceProviderFactory());
+
+		IServiceFactoryAdapter _serviceProviderFactory = new ServiceFactoryAdapter<IServiceCollection>(new MauiServiceProviderFactory(false));
+
 		bool _hostBuilt;
 		HostBuilderContext? _hostBuilderContext;
 		IHostEnvironment? _hostEnvironment;
 		IServiceProvider? _serviceProvider;
 		IServiceCollection? _services;
 		IConfiguration? _hostConfiguration;
 		IConfiguration? _appConfiguration;
-		App? _app;
 
 		public AppHostBuilder()
 		{
 
 		}
 		public IDictionary<object, object> Properties => new Dictionary<object, object>();
 
-		public IHost Build(IApp app)
+		public static IAppHostBuilder CreateDefaultAppBuilder()
```

Comment:
Good call @jsuarezruiz! When I saw this I smiled inside. Even though this technically needed to be moved from `App`, this made it so much more consistent with other things, like ASP.NET and also fits here way better. Now people can just do `AppHostBuilder.CreateDefaultAppBuilder()` without having a full app object.